irohad/validation/impl/stateful_validator_impl.cpp

```diff
@@ -138,10 +138,13 @@ namespace iroha {
                                *batch_parser_);
 
       // Since proposal came from ordering gate it was already validated.
-      // All transactions has been validated as well
+      // All transactions are validated as well
       // This allows for unsafe construction of proposal
-      validation_result->verified_proposal = factory_->unsafeCreateProposal(
-          proposal.height(), proposal.createdTime(), valid_txs);
+      validation_result->verified_proposal =
+          std::const_pointer_cast<const shared_model::interface::Proposal>(
```

Comment:
Maybe add a TODO here to refactor proposal factory with const pointer and remove const cast here?